www/src/pages/table/forms/TransformationForm.tsx

```diff
@@ -0,0 +1,284 @@
+
+const styles = {
+  table: {
+    "& div.ant-table-container": {
+      border: "1px solid #bbb",
+      borderRadius: 4,
+      borderLeft: "1px solid #bbb !important",
+    },
+    "& div.ant-card-body": {
+      paddingLeft: 0,
+      paddingRight: 0,
+    },
+    "& th": {
+      fontWeight: 600,
+    },
+  },
+};
+
+export interface TransformationFormProps {
+  type: string;
+  tableId: number;
+}
+
+export const TransformationForm = withStyles(styles)(
+  observer(
+    ({
+      classes,
+      tableId,
+    }: TransformationFormProps & WithStyles<typeof styles>) => {
+      const onDone = () => Modal.destroyAll();
+      const actionRef = useRef<ActionType>();
+      const [form] = Form.useForm();
+
+      const options = {
+        autoIndent: "full",
+        contextmenu: true,
+        fontFamily: "monospace",
+        fontSize: 13,
+        lineHeight: 24,
+        matchBrackets: "always",
+        minimap: {
+          enabled: false,
+        },
+        scrollbar: {
+          horizontalSliderSize: 4,
+          verticalSliderSize: 18,
+        },
+        selectOnLineNumbers: true,
+        roundedSelection: false,
+        readOnly: false,
+        cursorStyle: "line",
+        automaticLayout: true,
+      };
+
+      const { Option } = Select;
+      const [result, setResult] = useState<TransformationTable[] | undefined>();
+      const [editorData, setEditorData] = useState<string | undefined>();
+
+      const columns = [
+        { dataIndex: "row_id", title: "Row ID" },
+        { dataIndex: "previous_value", title: "Previous Value" },
+        {
+          dataIndex: "result",
+          title: "Result",
+          render: (transformed_value: string | boolean) => {
+            if (
+              typeof transformed_value === "string" &&
+              transformed_value.includes("Traceback")
+            ) {
+              return (
+                <>
+                  {" "}
+                  {
+                    <Tag color={"volcano"}>
+                      <pre>{transformed_value}</pre>
+                    </Tag>
+                  }
+                </>
+              );
+            } else {
+              return transformed_value;
+            }
+          },
+        },
+      ];
+
+      const onExecute = async () => {
+        const transformPayload = new TPayload();
+        transformPayload.type = form.getFieldValue("type");
+        transformPayload.code = editorData;
+        transformPayload.mode = "restrictedpython";
+        transformPayload.datapath = form.getFieldValue("datapath");
+        transformPayload.outputpath = form.getFieldValue("outputpath");
+        transformPayload.tolerance = form.getFieldValue("tolerance");
+        transformPayload.rows = form.getFieldValue("rows");
+
+        let response = await postData(tableId, transformPayload);
+        setResult(response);
+      };
+
+      function handleEditorChange(value: string | undefined) {
+        setEditorData(value);
+      }
```

Comment:
create a separated component for editor.

Reply:
created